components/table/ReportTable.tsx

```diff
@@ -57,44 +61,85 @@ function TotalHours({
     )
 }
 
-function CustomerHoursRow({
+function ProjectHoursRow({
     entries,
-    customer,
-}: CustomerHoursRowProps): JSX.Element {
+    project,
+}: ProjectHoursRowProps): JSX.Element {
     return (
         <ListItem>
-            Total hours for {customer.name}: {entriesQuantitySum(entries)}
+            Total hours for {project.name}: {entriesQuantitySum(entries)}
         </ListItem>
     )
 }
 
-function ProjectHoursRow({
+function CustomerHoursRow({
     entries,
-    project,
-}: ProjectHoursRowProps): JSX.Element {
+    customer,
+    projects,
+}: CustomerHoursRowProps): JSX.Element {
     return (
-        <ListItem>
-            Total hours for {project.name}: {entriesQuantitySum(entries)}
-        </ListItem>
+        <>
+            <ListItem>
+                Total hours for {customer.name}: {entriesQuantitySum(entries)}
+            </ListItem>
+            {
+                <UnorderedList>
+                    {projects.map(
+                        (project) =>
+                            project.id && (
+                                <ProjectHoursRow
+                                    entries={entries.filter(
+                                        (entry) =>
+                                            entry.timesheet.project?.id ===
+                                            project.id
+                                    )}
+                                    key={`project-hours-row-${project.id}`}
+                                    project={project}
+                                ></ProjectHoursRow>
+                            )
+                    )}
+                </UnorderedList>
+            }
+        </>
     )
 }
 
 function EmployeeHoursRow({
     entries,
     employee,
+    projects,
 }: EmployeeHoursRowProps): JSX.Element {
     return (
-        <ListItem>
-            Total hours for {employee.firstName} {employee.lastName}:{" "}
-            {entriesQuantitySum(entries)}
-        </ListItem>
+        <>
+            <ListItem>
+                Total hours for {employee.firstName} {employee.lastName}:{" "}
+                {entriesQuantitySum(entries)}
+            </ListItem>
+            <UnorderedList>
+                {projects.map(
+                    (project) =>
+                        project.id && (
+                            <ProjectHoursRow
+                                entries={entries.filter(
+                                    (entry) =>
+                                        entry.timesheet.project?.id ===
+                                        project.id
+                                )}
+                                key={`project-hours-row-${project.id}`}
```

Comment:
`key` should be unique. This key is repeated in `EmployeeHoursRow`.  You could make this `project-hours-row-employee-${project.id}`.